### PR TITLE
[in-proc backport] Replace build-extension.ps1 with SiteExtension.csproj

### PR DESCRIPTION
**Pull request by jviau**
### Issue describing the changes in this PR

This is the in-proc backport for #10168 

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Replaces `build-extension.ps1` with an msbuild project to build our site extension for net6.0 and net8.0 simultaneously.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -499,6 +499,4 @@ local.settings.json
 /tools/ExtensionsMetadataGenerator/test/ExtensionsMetadataGeneratorTests/runtimeAssemblies.txt
 
 local.settings.json
-msbuild.binlog
```
Comment (jviau) [inline .gitignore:502]: these are already in gitignore

**Review by liliankasem — APPROVED** [review]

**Comment by jviau** [comment]
Current blocker for this PR:

Part of this change was to leverage msbuild project dependencies to build in parallel and avoid double-building when we publish the separate net6 and net8 site extensions. A result of this is changing how the version property resolves: using TFM instead of a supplied parameter. This has led to an issue: `WebJobs.Script` is **not** multi-targeted like our other assemblies, meaning it is `4.37.0` when others are `4.{6|8}37.0`. And this has real impact as we use `WebJobs.Script`'s assembly version as the reported version everywhere, meaning we will report `4.37.0` with this change.

Fix options:

1. Multi-target `WebJobs.Script` to `net6/net8`
    - I think we should do this eventually as my testing shows it brings in extra analyzers, which are catching potentially real issues with that assembly.
2. Explicitly build each TFM and use old versioning logic.


I will go with option 2 for now, then follow up on option 1 in another PR.

**Review by satvu — APPROVED** [review]

File: build/python.props [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <Project>
-  <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.PythonWorker" Version="4.34.0" />
-  </ItemGroup>
+  <!-- See /eng/build/Workers.Python.props -->
+  <Target Name="_DoNotImport" BeforeTargets="Build">
+    <Error Text="This file should not be imported. Import '$(TargetsRoot)Workers.props'." />
```
Comment (hallvictoria) [inline build/python.props:4]: Should this be `Import '$(WorkersProps)'.`? Based off of what's currently in the dev branch

**Review by hallvictoria — APPROVED** [review]

File: build/python.props (hunk 2 above)
Comment (jviau) [reply build/python.props:4]: That change will come in a follow up PR to port from dev.

**Pull request merged**